src/main/java/slimeknights/tconstruct/blocks/CommonBlocks.java

```diff
@@ -1,127 +1,67 @@
-    BlockRegistryAdapter registry = new BlockRegistryAdapter(event.getRegistry());
-
-    // crafting related
-    registry.register(new Block(BlockProperties.GENERIC_SAND_BLOCK), "grout");
-
-    registry.register(new GraveyardSoilBlock(BlockProperties.GENERIC_SAND_BLOCK), "graveyard_soil");
-    registry.register(new ConsecratedSoilBlock(BlockProperties.GENERIC_SAND_BLOCK), "consecrated_soil");
-
-    registry.register(new SlimyMudBlock(BlockProperties.GENERIC_SAND_BLOCK, SlimyMudBlock.MudType.SLIMY_MUD_GREEN), "slimy_mud_green");
-    registry.register(new SlimyMudBlock(BlockProperties.GENERIC_SAND_BLOCK, SlimyMudBlock.MudType.SLIMY_MUD_BLUE), "slimy_mud_blue");
-    registry.register(new SlimyMudBlock(BlockProperties.GENERIC_SAND_BLOCK, SlimyMudBlock.MudType.SLIMY_MUD_MAGMA), "slimy_mud_magma");
+  private static final BlockDeferredRegister BLOCKS = new BlockDeferredRegister(TConstruct.modID);
+  private static final Item.Properties generalProps = new Item.Properties().group(TinkerRegistry.tabGeneral);
 
-    registry.registerSlabsAndStairs(new Block(BlockProperties.LAVAWOOD), "lavawood");
-    registry.registerSlabsAndStairs(new Block(BlockProperties.FIREWOOD), "firewood");
+  public static void init() {
+    IEventBus modEventBus = FMLJavaModLoadingContext.get().getModEventBus();
 
-    // Metal Blocks
-    registry.register(new BeaconBaseBlock(BlockProperties.GENERIC_METAL_BLOCK), "cobalt_block");
-    registry.register(new BeaconBaseBlock(BlockProperties.GENERIC_METAL_BLOCK), "ardite_block");
-    registry.register(new BeaconBaseBlock(BlockProperties.GENERIC_METAL_BLOCK), "manyullyn_block");
-    registry.register(new BeaconBaseBlock(BlockProperties.GENERIC_METAL_BLOCK), "knightslime_block");
-    registry.register(new BeaconBaseBlock(BlockProperties.GENERIC_METAL_BLOCK), "pigiron_block");
-    registry.register(new BeaconBaseBlock(BlockProperties.GENERIC_METAL_BLOCK), "alubrass_block");
-    registry.register(new BeaconBaseBlock(BlockProperties.GENERIC_GEM_BLOCK), "silky_jewel_block");
-
-    registry.register(new GlowBlock(BlockProperties.GLOW), "glow");
+    BLOCKS.register(modEventBus);
   }
 
-  @SubscribeEvent
-  static void registerBlockItems(final RegistryEvent.Register<Item> event) {
-    BlockItemRegistryAdapter registry = new BlockItemRegistryAdapter(event.getRegistry(), TinkerRegistry.tabGeneral);
-
-    registry.registerBlockItem(grout);
-
-    registry.registerBlockItem(graveyard_soil);
-    registry.registerBlockItem(consecrated_soil);
-
-    registry.registerBlockItem(slimy_mud_green);
-    registry.registerBlockItem(slimy_mud_blue);
-    registry.registerBlockItem(slimy_mud_magma);
-
-    registry.registerBlockItem(lavawood);
-    registry.registerBlockItem(firewood);
+  /* Crafting related items */
+  public static final BlockItemObject<Block> grout = BLOCKS.register("grout", () -> new Block(BlockProperties.GENERIC_SAND_BLOCK), (b) -> new BlockTooltipItem(b, generalProps));
```

Comment:
Since we use `BlockItemTooltip` everywhere, I think it might be worth making that the default in the registry. Alternatively, add a method `setDefaultBlockItem(Function<Block,? extends BlockItem> mapper)` to the block registry.

Reply:
I mean. we don't use it literally everywhere. Not all have a tooltip.

Reply:
Good point. So maybe name it `tooltipBlockItem` and pass it as needed.